core/dbt/node_runners.py

```diff
@@ -467,9 +467,13 @@ def execute_test(self, test):
         num_rows = len(table.rows)
         if num_rows > 1:
             num_cols = len(table.columns)
-            raise RuntimeError(
+            exc = RuntimeError(
```

Comment:
I think this should probably always result in an Exception. This indicates that the schema test itself is malformed, and is less a reflection of the quality of the data being tested.